### PR TITLE
Apim 5905 manual subscription federated api

### PR DESCRIPTION
**Pull request by michel-barret**
**Issue** [APIM-5905](https://gravitee.atlassian.net/browse/APIM-5905)

## Description

In some case we don't have federatedApiDefinition field in input of update.

So because it's impossible to don't have it we keep the previous one in this case.

[APIM-5905]: https://gravitee.atlassian.net/browse/APIM-5905?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-bqgvixiszc.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/UpdateFederatedApiUseCase.java [hunk 1]
```diff
@@ -49,4 +53,19 @@ public Output execute(Input input) {
...
+            currentApi
+                .toBuilder()
+                .name(newOne.getName())
+                .description(newOne.getDescription())
+                .version(newOne.getVersion())
+                .apiLifecycleState(newOne.getApiLifecycleState())
+                .visibility(newOne.getVisibility())
+                .labels(newOne.getLabels())
+                .categories(newOne.getCategories())
+                .build();
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/UpdateFederatedApiUseCase.java]: Groups are missing

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/domain_service/UpdateFederatedApiDomainServiceTest.java [hunk 2]
```diff
@@ -351,6 +356,50 @@ public void update_creates_index() {
             );
     }
 
+    @Disabled // TODO move to service
+    @Test
+    @DisplayName(
+        "In some case (specially when we come from UpdateApiFederated) we don't have federationApiDefinition, so we keep the previous"
+    )
+    public void dont_erase_federated_defenition() {
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/domain_service/UpdateFederatedApiDomainServiceTest.java]:
```suggestion
    public void dont_erase_federated_definition() {
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/use_case/UpdateFederatedApiUseCaseTest.java [hunk 3]
```diff
@@ -206,4 +212,94 @@ public void update_creates_index() {
...
+            .visibility(Api.Visibility.PRIVATE)
+            .lifecycleState(Api.LifecycleState.STOPPED)
+            .picture("input")
+            .groups(Set.of("input"))
+            .categories(Set.of("input"))
+            .labels(List.of("input"))
+            .disableMembershipNotifications(false)
+            .apiLifecycleState(Api.ApiLifecycleState.DEPRECATED)
+            .background("input")
+            .build();
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/use_case/UpdateFederatedApiUseCaseTest.java:245]: FYI, we have Fixtures classes like `fixtures.core.model.ApiFixtures` to help build models easily

Comment (michel-barret) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/use_case/UpdateFederatedApiUseCaseTest.java:245]: Yes but here I want set all fields exhaustively

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**